synapse/events/snapshot.py

```diff
@@ -135,6 +135,8 @@ class EventContext(UnpersistedEventContextBase):
     delta_ids: Optional[StateMap[str]] = None
     app_service: Optional[ApplicationService] = None
 
+    _state_map_before_event: Optional[StateMap[str]] = None
```

Comment:
This needs a comment in the docstring.

Reply:
I'm failing to see where this ever gets set when creating an `EventContext`.

Reply:
I think this is to bring EventContext into alignment with UnpersistedEventContext, from which it inherits and which has this field. There's also a separate fix I am working on that will use this field.

Reply:
(Note necessarily for this PR!)

Reply:
Actually -- should this be part of `UnpersistedEventContextBase` instead?

> There's also a separate fix I am working on that will use this field.

I think we should derfer the changes to this file until we're ready to use them, otherwise we might end up with dead code. (Sorry for the change of heart.)

Reply:
No problem!